tests/test_core.py

```diff
@@ -1236,15 +1236,16 @@ def test_exceptions_3(self):
     self.do_run('src.js', 'Caught exception: Hello\nDone.', args=['2'], no_build=True)
 
   def test_exceptions_allowed(self):
-    self.set_setting('DISABLE_EXCEPTION_CATCHING', 2)
```

Comment:
Shouldn't we set it to 0 instead and not delete the line? The same for other functions.

Reply:
No because we set `EXCEPTION_CATCHING_ALLOWED` below and setting them both at the same time is a warning (and we are building with `-Werror`)

Reply:
No I meant 0, not 1. The default setting for `DISABLE_EXCEPTION_CATCHING` is 1, so I thought we needed to turn it off to catch exceptions. Settings `DISABLE_EXCEPTION_CATCHING == 0` and `EXCEPTION_CATCHING_ALLOWED` at the same time is an error? 

Or does setting `EXCEPTION_CATCHING_ALLOWED` automatically imply `DISABLE_EXCEPTION_CATCHING == 0` or something?

Reply:
Yes exactly, setting EXCEPTION_CATCHING_ALLOWED implied DISABLE_EXCEPTION_CATCHING.

Also setting `DISABLE_EXCEPTION_CATCHING` to anything, including 0, at the same time as `EXCEPTION_CATCHING_ALLOWED` an error or warning (=2 is a warning =1, or =0 is an error).  They should never appear together .. although we need to continue to support =2 for backward compat (we give a warning in this case).

Reply:
I see. But why is setting `DISABLE_EXCEPTION_CATCHING` to 0 also an error? Are gonna do that internally anyway?

Reply:
It seems cleaner to me that they should never be specified together.  Don't you think they should be 100% mutually exclusive?    

That fact that internally we set `DISABLE_EXCEPTION_CATCHING` to zero when `EXCEPTION_CATCHING_ALLOWED` is specified is kind of an implementation detail I think.

Reply:
True, but `DISABLE_EXCEPTION_CATCHING` is still an exposed settings variable, so I think a warning can be fine, but an error seems weird, especially specifying both of them is semantically correct. Also this will not be backward compatible.